front-end/src/visualization/Edge.jsx

```diff
@@ -1,18 +1,36 @@
 import { Box } from '@chakra-ui/react';
+import './Visualization.css'
 
 const Edge = ({ from, to, color }) => {
-    const startX = from.x + 25;
-    const startY = from.y + 25;
-    const endX = to.x; 
-    const endY = to.y;
+    const nodeSize = 5 * 16;
+    const shadowPadding = 12;
+    const nodeRadius = nodeSize / 2 + shadowPadding;
 
-    const angle = Math.atan2(endY - startY, endX - startX) * 180 / Math.PI;
-    const length = Math.sqrt((endX - startX) ** 2 + (endY - startY) ** 2);
+    const startX = from.x + nodeSize / 2;
+    const startY = from.y + nodeSize / 2;
+    const endX = to.x + nodeSize / 2;
+    const endY = to.y + nodeSize / 2;
+
+    const dx = endX - startX;
+    const dy = endY - startY;
+    const distance = Math.sqrt(dx * dx + dy * dy);
+    const unitX = dx / distance;
+    const unitY = dy / distance;
+
+    const adjustedStartX = startX + unitX * nodeRadius;
+    const adjustedStartY = startY + unitY * nodeRadius;
+    const adjustedEndX = endX - unitX * nodeRadius;
+    const adjustedEndY = endY - unitY * nodeRadius;
+
+    const newDx = adjustedEndX - adjustedStartX;
+    const newDy = adjustedEndY - adjustedStartY;
+    const newLength = Math.sqrt(newDx * newDx + newDy * newDy);
+    const angle = Math.atan2(newDy, newDx) * 180 / Math.PI;
```

Comment:
Code is okay! Just making a comment. Can you run me explain me this code on our next 1:1? Please add it to our agenda :)